docs/site/Interceptors.md

```diff
@@ -0,0 +1,583 @@
+---
+lang: en
+title: 'Interceptors'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Interceptors.html
+---
+
+## Overview
+
+Interceptors are reusable functions to provide aspect-oriented logic around
+method invocations. There are many use cases for interceptors, such as:
+
+- Add extra logic before / after method invocation, for example, logging or
+  measuring method invocations.
+- Validate/transform arguments
+- Validate/transform return values
+- Catch/transform errors, for example, normalize error objects
+- Override the method invocation, for example, return from cache
+
+The following diagram illustrates how interceptors can be applied to the
+invocation of a method on the controller class.
+
+![Interceptors](imgs/interceptors.png)
+
+## Basic use
+
+### Interceptors on controllers
+
+Interceptors are supported for public controller methods (including both static
+and prototype) and handler functions for REST routes.
+
+Controller methods decorated with `@intercept` are invoked with applied
+interceptors for corresponding routes upon API requests.
+
+```ts
+import {intercept} from '@loopback/context';
+
+@intercept(log)
+export class OrderController {
+  @intercept('caching-interceptor')
+  async listOrders(userId: string) {
+    // ...
+  }
+}
+```
+
+It's also possible to configure global interceptors that are invoked before
+method level interceptors. For example, the following code registers a global
+`caching-interceptor` for all methods.
+
+```ts
+app
+  .bind('caching-interceptor')
+  .toProvider(CachingInterceptorProvider)
+  .apply(asInterceptor);
+```
+
+Global interceptors are also executed for route handler functions without a
+controller class. See an example in
+[Route Handler](Routes.md#using-partial-openapi-spec-fragments).
+
+### Create a proxy to apply interceptors
+
+A
+[proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy)
+can be created from the target class or object to apply interceptors. This is
+useful for the case that a controller declares dependencies of repositories or
+services and would like to allow repository or service methods to be
```

Comment:
@raymondfeng Is the interception limited to the instance in the controller or is it a global effect?